libweston/backend-rdp/rdp.c

```diff
@@ -1470,11 +1472,71 @@ rdp_validate_button_state(RdpPeerContext *peerContext, bool pressed, uint32_t* b
 	return;
 }
 
+static bool
+rdp_notify_wheel_scroll(RdpPeerContext *peerContext, UINT16 flags, uint32_t axis)
+{
+	struct weston_pointer_axis_event weston_event;
+	struct rdp_backend *b = peerContext->rdpBackend;
+	int ivalue;
+	double value;
+	struct timespec time;
+
+	/*
+		* The RDP specs says the lower bits of flags contains the "the number of rotation
+		* units the mouse wheel was rotated".
+		*
+		* https://blogs.msdn.microsoft.com/oldnewthing/20130123-00/?p=5473 explains the 120 value
+		*/
+	if (flags & PTR_FLAGS_WHEEL_NEGATIVE)
+		ivalue = (int)((char)(flags & 0xff));
+	else
+		ivalue = (flags & 0xff);
+
+	/*
+		* Flip the scroll direction as the RDP direction is inverse of X/Wayland 
+		* for vertical scroll 
+		*/
+	if (axis == WL_POINTER_AXIS_VERTICAL_SCROLL)
+		ivalue *= -1;
+
+	/*
+		* Accumulate the wheel increments.
+		*
+		* Every 12 wheel increments, we will send an update to our Wayland
+		* clients with an updated value for the wheel for smooth scrolling.
+		*
+		* Every 120 wheel increments, we tick one discrete wheel click.
+		*/
+	peerContext->accumWheelRotationPrecise += ivalue;
```

Comment:
I think these need to be accumulated per axis.

Reply:
Follow up change can be found here:

https://github.com/microsoft/weston-mirror/pull/28